### PR TITLE
Added mapping functions to `Sequence` as alternatives to `KeyPath`

### PR DESCRIPTION
**Pull request by MartonioJunior**
🚀

Methods with mapping alternatives: `sorted(by:)`, `sorted(by:with:)`, `sorted(by:and:)`, `sorted(by:and:and:)`, `sum(for:)`, `first(where:equals:)`. All of this keeps compatibility as passing a `KeyPath` will make the code opt for the KeyPath version of the method. (Standalone PR as requested on #1167)

## Checklist

- [X] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [X] New extensions are written in Swift 5.6.
- [X] New extensions support iOS 12.0+ / tvOS 12.0+ / macOS 10.13+ / watchOS 4.0+, or use `@available` if not.
- [X] I have added tests for new extensions, and they passed.
- [X] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [X] All extensions are declared as **public**.
- [X] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,11 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - Added `+=`, `-=`, `*=`, `/=` to add, subtract, multiply and divide measurements. [#1162](https://github.com/SwifterSwift/SwifterSwift/pull/1162) by [Roman Podymov](https://github.com/RomanPodymov)
 - **UIView**
   - Added `removeBlur()` method for removing the applied blur effect from the view. [#1159](https://github.com/SwifterSwift/SwifterSwift/pull/1159) by [regi93](https://github.com/regi93)
+  - Added `makeCircle(diameter:)` method to make the view circular. [#1165](https://github.com/SwifterSwift/SwifterSwift/pull/1165) by [happyduck-git](https://github.com/happyduck-git)
```
Comment (guykogus) [inline CHANGELOG.md:15]: Why are these included?

File: Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift [hunk 2]
```diff
@@ -237,6 +302,18 @@ public extension Sequence {
...
+
+    /// SwifterSwift: Returns the first element of the sequence with having property by given key path equals to given
+    /// `value`.
+    ///
+    /// - Parameters:
+    ///   - map: Function for `Element` to compare.
+    ///   - value: The value to compare with `Element` property.
+    /// - Returns: The first element of the collection that has property by given map function equals to given `value` or
+    /// `nil` if there is no such element.
+    func first<T: Equatable>(where map: (Element) throws -> T, equals value: T) rethrows -> Element? {
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift:314]: I love that you made this function throwable. Do you want to do that for the other new functions you created?

File: Tests/UIKitTests/UIViewExtensionsTests.swift [hunk 3]
```diff
@@ -88,6 +88,14 @@ final class UIViewExtensionsTests: XCTestCase {
         XCTAssertEqual(view.layer.mask?.bounds, shape.bounds)
         XCTAssertEqual(view.layer.mask?.cornerRadius, shape.cornerRadius)
     }
+    
+    func testMakeCircle() {
```
Comment (guykogus) [inline Tests/UIKitTests/UIViewExtensionsTests.swift:92]: Shouldn't be here too

**Review by guykogus — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (MartonioJunior) [reply CHANGELOG.md:15]: I think this and the 3rd change request was due to my fork being synced before #1165 was merged. I'll look into it soon.

File: Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift (hunk 2 above)
Comment (MartonioJunior) [reply Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift:314]: Oh, for sure!

**Comment by MartonioJunior** [comment]
Made the requested changes:
- Added throws/rethrows to the new methods
- Adjusted branch to only have the commit related to the implementation

File: Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift [hunk 4]
```diff
@@ -215,6 +248,27 @@ public extension Sequence {
...
+    func sorted<T: Comparable, U: Comparable, V: Comparable>(by map1: (Element) throws -> T,
+                                                             and map2: (Element) throws -> U,
+                                                             and map3: (Element) throws -> V) rethrows -> [Element] {
+        return try sorted {
+            if try map1($0) != map1($1) {
+                return try map1($0) < map1($1)
+            }
+            if try map2($0) != map2($1) {
+                return try map2($0) < map2($1)
+            }
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift:267]: Same comment about repeated calculations.
```suggestion
            let value10 = map1($0)
            let value11 = map1($1)
            if try value10 != value11 {
                return try value10 < value11
            }
            
            let value20 = map2($0)
            let value21 = map2($1)
            if try value20 != value21 {
                return try value20 < value21
            }
```

**Review by guykogus — CHANGES_REQUESTED** [review]

**Comment by guykogus** [comment]
Sorry for reviewing again after so long. Also, please note the conflicts.

File: Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift [hunk 5]
```diff
@@ -194,7 +211,23 @@ public extension Sequence {
...
+                                              and map2: (Element) throws -> U) rethrows -> [Element] {
+        return try sorted {
+            if try map1($0) != map1($1) {
+                return try map1($0) < map1($1)
+            }
+            return try map2($0) < map2($1)
+        }
+    }
+
+    /// SwifterSwift: Returns a sorted sequence based on three map functions. Whenever the values of one map function match, the
```
[the remark this answers is not in the document]
Comment (MartonioJunior) [reply Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift]: Fixed it

File: CHANGELOG.md [hunk 6]
```diff
@@ -17,6 +17,10 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - Added `removeBlur()` method for removing the applied blur effect from the view. [#1159](https://github.com/SwifterSwift/SwifterSwift/pull/1159) by [regi93](https://github.com/regi93)
   - Added `makeCircle(diameter:)` method to make the view circular. [#1165](https://github.com/SwifterSwift/SwifterSwift/pull/1165) by [happyduck-git](https://github.com/happyduck-git)
 
+### Changed
```
[the remark this answers is not in the document]
Comment (MartonioJunior) [reply CHANGELOG.md:32]: Updated Changelog to address this

File: .github/workflows/CI.yml [hunk 7]
```diff
@@ -11,7 +11,7 @@ on:
 jobs:
   Darwin:
     name: Darwin
-    runs-on: macos-13
```
Comment (LucianoPAlmeida) [inline .github/workflows/CI.yml:14]: Something is wrong with this diff, seems like a bad merge incorrect rebase

**Comment by guykogus** [comment]
@MartonioJunior there seems to be an issue here with a recent merge. You have way too many changed files.

**Comment by MartonioJunior** [comment]
> @MartonioJunior there seems to be an issue here with a recent merge. You have way too many changed files.

I'll take a look as soon as I can, this was likely an issue caused by how I rebased the branch to add in the most recent commits from the master branch.

**Comment by MartonioJunior** [comment]
Tthere was a merge in the branch that had an earlier master commit as the basis, hence why it was appearing on the PR.

I also updated with the new commit, so it should be fine now, but if I still missed something, feel free to fire a comment.

**Review by guykogus — APPROVED** [review]
